src/ui/Combobox/ComboboxInput.ts

```diff
@@ -40,9 +40,9 @@ export class ComboboxInput {
     if (!this.combobox.options.clearOnBlur) {
```

Comment:
How does the keyboard interaction work on android to test the changes with TalkBack? Is it possible to use the cursor keys on the virtual keyboard somehow? or is it by swiping up/down in a certain way? or by clicking a physical button on the device?

Reply:
Android's TalkBack supports multiple different ways to focus elements.

The first one is by swiping, which doesn't appear to trigger the `focus`, `blur`, focusin` and `focusout` events (or at-least not on Samsung's implementation). This also doesn't seem to apply the `focused` style or be detectable by `querySelector`. A user can swipe left to focus the previous item and swipe right to focus the next item. A user can also swipe from the status bar to the element they wish to focus. An element can then be activated by double-tapping anywhere on the screen, which triggers the `click` event. TalkBack also allows users to change what type of element they wish to focus, meaning they can skip over multiple elements with a single left/right swipe.

The second one is with a keyboard plugged into the device. This can be through Bluetooth, through USB or even through the emulator using the host's keyboard. I think this also works with Chrome's remote devtools. With a keyboard, everything works exactly as it does on a computer, apart from Shift + Tab which seems not to work.

Using one of those two methods after the other appears to remove the focus from the other, even though they seem to use a different kind of focus. When switching from one to the other, it looks like the focus resumes from where it last was when using the same method. Meaning that if you press Tab until you reach the search bar, swiping right will focus the first element in the page. Pressing tab again should focus the element after the search bar, independently of how many times you swiped right. I'll test this to confirm it.

Reply:
Good job on the investigation 👍 
It's not easy...